src/sagemaker/jumpstart/utils.py

```diff
@@ -752,16 +773,25 @@ def is_valid_model_id(
     s3_client = sagemaker_session.s3_client if sagemaker_session else None
     region = region or constants.JUMPSTART_DEFAULT_REGION_NAME
     model_version = model_version or "*"
-
     models_manifest_list = accessors.JumpStartModelsAccessor._get_manifest(
-        region=region, s3_client=s3_client
+        region=region, s3_client=s3_client, model_type=enums.JumpStartModelType.OPENSOURCE
+    )
+    open_source_model_id_set = {
+        model.model_id for model in models_manifest_list
+    }
+
+    proprietary_manifest_list = (
```

Comment:
out of scope for this PR, but perhaps it is better to store this structure as a dict, and then if you ever need a list, call `.values()` on the dict.